### PR TITLE
feat(announcement-modal): Backend add announcementVersion user metadata

### PR DESCRIPTION
**Pull request by theMogget**
## Problem

We need to track the latest announcement a user has seen, and display any later announcements.

re: #798 

## Solution
**Features**:
* Add user_features table model
* Add UserFeatures to User table model
* Edit `/settings` to also return announcement_version
* Create new `/settings/announcement-version` endpoint to update user's announcement_version

### CODE REVIEW

File: backend/src/core/models/user/user-features.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  @Column({
+    type: DataType.INTEGER,
+    allowNull: false,
+    primaryKey: true,
+  })
+  userId!: number
+
+  @Default(null)
+  @Column({
+    type: DataType.INTEGER,
```
Comment (jeantanzj) [inline backend/src/core/models/user/user-features.ts:23]: string?  the announcementVersion should probably follow the package version (https://semver.org/)

File: backend/src/core/models/user/user.ts [hunk 2]
```diff
@@ -35,6 +36,9 @@ export class User extends Model<User> {
   @HasOne(() => UserDemo)
   demo!: UserDemo
 
+  @HasOne(() => UserFeatures)
+  userFeatures!: UserFeatures
```
Comment (jeantanzj) [inline backend/src/core/models/user/user.ts:40]: rename this too

File: backend/src/core/routes/settings.routes.ts [hunk 3]
```diff
@@ -27,6 +27,12 @@ const demoDisplayedValidator = {
   }),
 }
 
+const updateAnnouncementVersionValidator = {
+  [Segments.BODY]: Joi.object({
+    announcement_version: Joi.number().required(),
```
Comment (jeantanzj) [inline backend/src/core/routes/settings.routes.ts:32]: type also has to change here

File: backend/src/core/services/credential.service.ts [hunk 4]
```diff
@@ -250,6 +261,7 @@ const getUserSettings = async (
       hasApiKey: !!user.apiKey,
       creds: user.creds,
       demo: user.demo,
+      userFeatures: user.userFeatures,
```
Comment (jeantanzj) [inline backend/src/core/services/credential.service.ts:264]: basically rename things across all the files where they should be renamed

**Review by jeantanzj — CHANGES_REQUESTED** [review]

File: backend/src/core/models/user/user-features.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  DataType,
+  Model,
+  Table,
+  ForeignKey,
+  Default,
+} from 'sequelize-typescript'
+import { User } from './user'
+
+@Table({ tableName: 'user_features', underscored: true, timestamps: true })
+export class UserFeatures extends Model<UserFeatures> {
```
[the remark this answers is not in the document]
Comment (theMogget) [reply backend/src/core/models/user/user-features.ts]: Done!

File: backend/src/core/models/user/user-features.ts (hunk 1 above)
Comment (theMogget) [reply backend/src/core/models/user/user-features.ts]: Done~

**Pull request merged**